src/app/common/httpclient.ts

```diff
@@ -108,68 +99,67 @@ export class HttpClient {
         //
         // Set Access-Token
-        req.headers.set('Access-Token', 'Bearer ' + this._conn.accessToken);
-        
-        return this._http.request(req).toPromise()
-            .catch(e => {
+        req.headers.set('Access-Token', 'Bearer ' + conn.accessToken);
+        return this._http.request(req)
+            .catch((e, _) => {
                 // Status code 0 possible causes:
                 // Untrusted certificate
                 // Windows auth, prevents CORS headers from being accessed
                 // Service not responding
-                if (e.status == 0) {
-                    //
-                    // The first request to the API fails because windows auth has not started yet.
-                    // We repeat the request because in this case the next request will succeed.
-                    return this._http.request(req).toPromise()
-                        .catch(err => {
-                            //
-                            // Check to see if connected
-                            return this._http.options(this._conn.url).toPromise()
-                                .catch(e => {
-                                    this._connectSvc.reconnect();
-                                    return Promise.reject("Not connected");
+                if (e instanceof Response) {
+                    if (e.status == 0) {
+                        return this._http.request(req)
+                            .catch((err, _) => {
+                                // Check to see if connected
+                                return this._http.options(this._conn.url)
+                                    .catch((e, _) => {
+                                        this._connectSvc.reconnect();
+                                        this.handleHttpError(err);
+                                        return Observable.throw(e)
+                                    })
                                 })
-                                .then(r => {
-                                    return this.handleHttpError(err);
-                                });
-                        });
+                    }
+                    this.handleHttpError(e, warn);
+                    return Observable.throw(e)
                 }
-                return this.handleHttpError(e, warn);
-            });
+                let apiError = <ApiError>({
+                    title: "unknown error",
+                    detail: JSON.stringify(e),
+                })
+                this._notificationService.apiError(apiError)
+                return Observable.throw(apiError)
+            })
     }
 
-    private handleHttpError(err, warn?: boolean): Promise<any> {
+    private handleHttpError(err, warn?: boolean) {
         if (err.status == 403 && err.headers.get("WWW-Authenticate") === "Bearer") {
             this._connectSvc.reconnect();
-            return Promise.reject("Not connected");
+            throw "Not connected"
```

Comment:
I guess this should Observable.throw. Would you test this scenarios and confirm which one is better between throw and Observable.throw?